data_structures/src/mainnet_validations.rs

```diff
@@ -16,6 +18,64 @@ pub const FIRST_HARD_FORK: Epoch = 192000;
 /// 28 April 2021 @ 9:00:00 UTC
 pub const SECOND_HARD_FORK: Epoch = 376320;
 
+/// Struct that count the positives votes of a WIP
+#[derive(Debug, Default, Clone, Serialize, Deserialize, PartialEq)]
+pub struct BitVotesCounter {
+    pub votes: u32,
+    pub period: u32,
+    pub wip: String,
+    pub init: u32,
+    pub end: u32,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, PartialEq)]
+pub struct BitTapiCounter {
+    pub info: HashMap<i8, Vec<BitVotesCounter>>,
```

Comment:
Same comment as before, since the key of this map is a bit index between 0 and 31, it can be replaced by an array of 32 elements:

```suggestion
    pub info: [Vec<BitVotesCounter>; 32],
```